### ISSUE TITLE
[JENKINS-68240] Add BuildNumber as response when a Job is submitted using Jenkins API 

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#9456]
<p>We have a use-case where multiple users can submit requests and our backend needs to process all of them in parallel and continuously check for status of the running job. We did some research on how to associate a job which is submitted with the buildnumber, currently we are using nextBuildnumber concept inorder to get build number and check the status, but this might not work when multiple users are sending requests at the same time. Please let us know if there is any way to achieve this and if I need to make a feature request to get the build number when a job is submitted</p>

---
<details><summary><i>Originally reported by <a title="Shravani Ravinuthala" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=shravani777">shravani777</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-68240" target="_blank">Add BuildNumber as response when a Job is submitted using Jenkins API </a></i></summary>
<i><ul>
<li><b>reported by</b>: shravani777
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2022-04-23T11:03:29+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwinter69">mwinter69</a>:</i>
<p>At the point in time when the job is started the build number is not known. There might be a queue in Jenkins, now imagine user A start the job, then user B and finally user C. Now user A decides to cancel his job waiting in the queue. If you returned the build number at the time it was submitted to the queue the numbers for jobs submitted by B and C would be wrong.</p>

<p>But there is a solution for this when you submit a job you get the ID as part of the response. You can now query the queue for this id</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre><jenkinsurl>/queue/item/<id>/api/xml</pre>
</div></div>
<p>Once the job started you will find in the api information about the execution of the job with the (build)number for a few minutes</p>

<p> </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
<leftItem _class=<span class="code-quote">"hudson.model.Queue$LeftItem"</span>>
...
<id>10</id>
...
<executable _class=<span class="code-quote">"hudson.model.FreeStyleBuild"</span>>
  <number>1</number>
  <url>http:<span class="code-comment">//localhost:8080/job/demo/1/</url>
</span></executable>
</leftItem>
</pre>
</div></div>
<p>In the api of the job itself you will see the field queueId</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
<freeStyleBuild _class=<span class="code-quote">"hudson.model.FreeStyleBuild"</span>>
...
<queueId>10</queueId>
...
</freeStyleBuild>
</pre>
</div></div>
<p> </p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER142714">JIRAUSER142714</a>:</i>
<p>Thank you very much <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwinter69" class="user-hover" rel="mwinter69">mwinter69</a>.  Is there any estimate on how much time it might be available exactly, because we checked it during the process and it looked like it was there. But after the end of the job it is saying NOT Found. So, just wanted to confirm if it will be available for as long as the job is running?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwinter69">mwinter69</a>:</i>
<p>Things will stay in the queue for about 5 minutes after the job started afaik, independent of how long the job takes to complete.</p>
